src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,648 @@
+                                   'high_frequency', 'entity_type'])
+
+# Define FSS Protection Point, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'height_agl', 'max_gain_dbi', 'pointing_azimuth',
+# 'pointing_elevation' 
+FssProtectionPoint = namedtuple('FssProtectionPoint',
+                                   ['latitude', 'longitude',
+                                    'height_agl', 'max_gain_dbi',
+                                    'pointing_azimuth', 'pointing_elevation'])
+
+
+# Define ESC information, i.e., a tuple with named fields of
+# 'antenna_height', 'antenna_azimuth', 'antenna_gain', 'antenna_pattern_gain'
+EscInformation = namedtuple('EscInformation',
+                                 ['antenna_height', 'antenna_azimuth',
+                                  'antenna_gain', 'antenna_pattern_gain'])
+
+def dbToLinear(x):
+  """This function returns dBm to mW converted value"""
+  return 10**(x / 10)
+
+def linearToDb(x):
+  """This function returns mW to dBm converted value"""
+  return 10 * np.log10(x)
+
+def getProtectedChannels(low_freq_hz, high_freq_hz):
+  """Gets protected channels list 
+
+  Performs 5MHz IAP channelization and returns a list of tuple containing 
+  (low_freq,high_freq)
+
+  Args:
+    low_freq_hz: Low frequency of the protected entity(Hz).
+    high_freq_hz: High frequency of the protected entity(Hz)
+  Returns:
+    An array of protected channel frequency range tuple 
+    (low_freq_hz,high_freq_hz). 
+  """
+  assert low_freq_hz < high_freq_hz, 'Low frequency is greater than high frequency'
+
+  channels = np.arange( max(low_freq_hz, 3550*MHZ), min(high_freq_hz, 3700*MHZ), 5*MHZ)
+
+  return [(low, high) for low,high in zip(channels, channels+5*MHZ)]
+
+
+def findGrantsInsideNeighborhood(grants, protection_point, entity_type):
+  """Finds grants inside protection entity neighborhood
+
+  Args:
+    grants: a list of grants of type GAA
+    protection_point: tuple containing (latitude,longitude) of a protection entity 
+    entity_type: enum of type ProtectedEntityType 
+  Returns:
+    grants_inside: a list of grants, each one being a namedtuple of type
+                   CbsdGrantInformation, of all CBSDs inside the neighborhood 
+                   of the protection constraint.
+  """
+  # Initialize an empty list
+  grants_inside = []
+
+  # Loop over each CBSD grant
+  for grant in grants:
+    # Compute distance from CBSD location to protection constraint location
+    dist_km, _, _ = vincenty.GeodesicDistanceBearing(grant.latitude, 
+                      grant.longitude, protection_point[0], protection_point[1])
+
+    # Check if CBSD is inside the neighborhood of protection constraint
+    if dist_km <= _DISTANCE_PER_PROTECTION_TYPE[entity_type][grant.cbsd_category == 'B']:
+      grants_inside.append(grant)
+
+  return grants_inside
+
+
+def findOverlappingGrants(grants, constraint):
+  """Finds grants inside protection entity neighborhood
```

Comment:
Please update the docstring description. It is identical to previous function, altthough it does things diferently.

Reply:
Updated   doc string